denops/ddu-devicon/main.ts

```diff
@@ -38,26 +39,32 @@ export function getSpecificFileIconData(
   specificFileIcons: Record<string, IconData>,
   isFolder?: boolean,
 ): IconData | undefined {
-  if (isFolder) return undefined
-  const filename = path.match(/[^\/]+$/)
-  if (filename != null) {
-    return specificFileIcons[filename[0]] ?? undefined
+  if (isFolder) {
+    return undefined;
   }
+  const filename = basename(path);
+  return specificFileIcons[filename] ?? undefined;
 }
 
 export function getCustomIconData(
   path: string,
   customIconData: Record<string, IconData>,
   isFolder?: boolean,
 ): IconData | undefined {
-  if (isFolder) return undefined
+  if (isFolder) {
+    return undefined;
+  }
   const def = getDeviconDef(path, isFolder);
-  const ext = path.replace(/(.*)\./, "");
-  if (!customIconData[ext]) return undefined
-  return {
-    icon: customIconData[ext].icon ?? (def?.icon ?? ''),
-    hl_group: customIconData[ext].hl_group ?? (def && getHighlightGroup(def))
+  const ext = (extname(path) === "")
+    ? basename(path)
+    : extname(path).slice(1, extname(path).length);
```

Comment:
```typescript
  let ext = extname(path).slice(1);
  if (ext === "") {
    ext = basename(path);
  }
```

Reply:
The call to extname should only need to be made once.
By the way, do we need a fallback to basename?